include/deal.II/meshworker/simple.h

```diff
@@ -211,16 +202,18 @@ namespace MeshWorker
                     const DOFINFO &info2);
     private:
       /**
-       * Assemble a single matrix into #matrix.
+       * Assemble a single matrix <code>M</code> into the element
+       * at <code>index</code> in the vector #matrix.
        */
       void assemble(const FullMatrix<double> &M,
+                    const unsigned int index,
                     const std::vector<types::global_dof_index> &i1,
                     const std::vector<types::global_dof_index> &i2);
```

Comment:
What about also supplying an <code>assemble</code> call without the <code>index</code> variable that only operates on the first matrix? Otherwise it is a bit awkward to have a specialization for <code>initialize</code> that takes a matrix reference but no matching spezialiation of assemble.